tracee/tracee.go

```diff
@@ -812,6 +828,38 @@ func (t *Tracee) handleError(err error) {
 
 // shouldProcessEvent decides whether or not to drop an event before further processing it
 func (t *Tracee) shouldProcessEvent(e RawEvent) bool {
+	if t.config.Filter.ArgFilter.Enabled {
+		// For each argument, check argument filter match
+		for _, tag := range e.ArgsTags {
+			argMeta, ok := t.DecParamName[e.Ctx.EventID%2][tag]
+			if !ok {
+				continue
+			}
+			key := ArgFilterKey{
+				EventId: e.Ctx.EventID,
+				ArgName: argMeta.Name,
+			}
+			if argFilter, ok := t.config.Filter.ArgFilter.Filters[key]; ok {
+				argVal := fmt.Sprint(e.RawArgs[tag])
```

Comment:
not a clocker, but I have a feeling that Sprint will be much less efficient than type assertion. I normally wouldn't care but since this is happening for every filter check, maybe better to replace.

Reply:
I know, but assertion will require to handle all argument types differently, and I didn't want to get into it now